packages/vite/src/node/plugins/importMetaGlob.ts

```diff
@@ -246,17 +246,30 @@ export async function parseImportGlob(
 
         if (!(name in knownOptions)) throw err(`Unknown options ${name}`)
 
-        if (property.value.type !== 'Literal')
+        const isTemplateLiteral = property.value.type === 'TemplateLiteral'
+        if (property.value.type !== 'Literal' && !isTemplateLiteral)
           throw err('Could only use literals')
 
-        const valueType = typeof property.value.value
+        if (
+          isTemplateLiteral &&
+          (property.value as TemplateLiteral).expressions.length > 0
+        ) {
+          throw err(
+            `Expected glob to be a string, but got dynamic template literal`
+          )
```

Comment:
Maybe we should take this opportunity to provide a nicer error message, e.g. what option that causes this error.
```suggestion
          throw err(
            `Expected option "${name}" to be a string, but got dynamic template literal`
          )
```

Reply:
yeah, this is nicer.